File: scala/src/main/scala/SimpleApp.scala
import org.apache.spark.sql.SparkSession

object SimpleApp {
  def main(args: Array[String]) {
      println("\n---------------------hello spark scala----------------------- ")

      val logFile = "/Users/h/Documents/spark/scala/README.md" 
      val spark = SparkSession.builder.appName("Simple Application").getOrCreate()
      val logData = spark.read.textFile(logFile).cache()
      // val numAs = logData.filter(line => line.contains("a")).count()
      // val numBs = logData.filter(line => line.contains("b")).count()
      // println(s"Lines with a: $numAs, Lines with b: $numBs")

      println("---------------------done1-----------------------\n");

  }
}

File: java/src/main/java/SimpleApp.java
import org.apache.spark.sql.SparkSession;
import org.apache.spark.sql.Dataset;

public class SimpleApp {
  public static void main(String[] args) {
    System.out.println("\n---------------------hello spark----------------------- ");

    //String logFile = "/usr/local/README.md"; // Should be some file on your system
    String logFile = "/Users/h/Documents/spark/java/README.md"; // Should be some file on your system


    SparkSession spark = SparkSession.builder().appName("Simple Application").getOrCreate();
    Dataset<String> logData = spark.read().textFile(logFile).cache();

    //long numAs = logData.filter(s -> s.contains("a")).count();
    // long numBs = logData.filter(s -> s.contains("b")).count();

    // System.out.println("Lines with a: " + numAs + ", lines with b: " + numBs);

    // spark.stop();

    System.out.println("---------------------done-----------------------\n");

  }
}
